pkg/generate/code/compare.go

```diff
@@ -562,7 +562,9 @@ func CompareStruct(
 		secondResAdaptedVarName := secondResVarName + "." + memberNameClean
 
 		var compareConfig *ackgenconfig.CompareFieldConfig
-		fieldConfig := fieldConfigs[memberFieldPath]
+		// memberFieldPath contains the field path along with the prefix "Spec." hence we
+		// would need to substring to exclude "Spec." to get correct field config.
```

Comment:
I see... A more future proof way of doing this would be to remove the length of the [`cfg.PrefixConfig.SpecField`](https://github.com/aws-controllers-k8s/code-generator/blob/386dc0d66152f1779e4bb315d153be5412361123/pkg/generate/config/config.go#L71) value. For targets like Crossplane, they use [`Spec.ForProvider`](https://github.com/aws-controllers-k8s/code-generator/blob/386dc0d66152f1779e4bb315d153be5412361123/pkg/generate/crossplane/config.go#L21) instead of `Spec`, which is why this configuration field exists. :)

Reply:
Thanks! updated.